### ISSUE TITLE
Tests failing with rails 3.2

### ISSUE THREAD

**Issue by avit** [grosser/test_after_commit#45]
Updating from 1.0.0 to 1.1.0 causes previously passing tests to fail under rails 3.2.

**Comment by avit** [comment]
It looks like the recent update to `@_current_transaction_records.clear` prevents this scenario from working:

```
describe "an after_commit hook" do
  let(:reservation) { Reservation.new(status: "reserved") }

  before do
    reservation.save!
  end

  it "should trigger on cancellation" do
    TestAfterCommit.with_commits(true) do
      expect(reservation).to receive(:send_cancel_message)
      reservation.cancel!
    end
  end
end
```

Moving the `save!` action inside the `TestAfterCommit` block makes it work. 

It used to be possible to inherit state from an outer context before enabling the commit hooks in the test. Is this a bug or intended change?

**Comment by grosser** [comment]
save in that case is outside of the after-commit block, so I'd expect it's
after-commit hooks to not fire ... if that was your question

**Comment by avit** [comment]
Originally I was implying that the `cancel!` would save and trigger a commit callback. I've changed the example code above to be explicit... 

Right now I can't reproduce this in a "rails new" test app, and I've worked around it in our app code by refactoring the original tests that were affected. I'm not sure if it was some interaction with FactoryGirl or the state machine gem that would save things strangely, but I can't make it happen in an isolated case. Let's close this.
